ca/certificate-authority.go

```diff
@@ -49,6 +51,71 @@ var badSignatureAlgorithms = map[x509.SignatureAlgorithm]bool{
 	x509.ECDSAWithSHA1:             true,
 }
 
+// Miscellaneous PKIX OIDs that we need to refer to
+var oidSubjectAltName = asn1.ObjectIdentifier{2, 5, 29, 17}
+var oidExtensionRequest = asn1.ObjectIdentifier{1, 2, 840, 113549, 1, 9, 14}
+
+// OID and fixed value for the "must staple" variant of the TLS Feature
+// extension:
+//
+//  Features ::= SEQUENCE OF INTEGER                  [RFC7633]
+//  enum { ... status_request(5) ...} ExtensionType;  [RFC6066]
+//
+// DER Encoding:
+//  30 03 - SEQUENCE (3 octets)
+//  |-- 02 01 - INTEGER (1 octet)
+//  |   |-- 05 - 5
+var oidTLSFeature = asn1.ObjectIdentifier{1, 3, 6, 1, 5, 5, 7, 1, 24}
+var mustStapleFeatureValue = []byte{0x30, 0x03, 0x02, 0x01, 0x05}
+var mustStapleExtension = signer.Extension{
+	ID:       cfsslConfig.OID(oidTLSFeature),
+	Critical: false,
+	Value:    hex.EncodeToString(mustStapleFeatureValue),
+}
+
+// Extract supported extensions from a CSR.  The following extensions are
+// currently supported:
+//
+// * 1.3.6.1.5.5.7.1.24 - TLS Feature [RFC7633], with the "must staple" value
+//
+// Other requested extensions are silently ignored.
+func extensionsFromCSR(csr *x509.CertificateRequest) (extensions []signer.Extension, err error) {
```

Comment:
Sorry, should have pointed this out earlier: This method is too long for naked returns. Please un-name the return variables, and do explicit returns instead.